scRNA-seq/01-normalizing_scRNA-seq.Rmd

```diff
@@ -61,28 +62,222 @@ and could use these same steps to filter and normalize the data.
-  tibble::column_to_rownames("gene") 
+sc_data <- readr::read_tsv(file.path("data", "unfiltered_darmanis_counts.tsv"), 
+                           progress = FALSE)
+
+# Read in the metadata
+sc_metadata <- readr::read_tsv(file.path("data", "unfiltered_darmanis_metadata.tsv"), 
+                               progress = FALSE)
+```
+
+## Filter the counts matrix
+  
+Single-cell RNA-seq experiments are generally much larger dimensions than bulk 
+RNA-seq, since every individual cell is being measured. 
+However, due to the high risk of technical artifacts, we don't necessarily want 
+to keep the data for all the cells and genes. 
+
+There are different approaches to filtering your data. 
+In an ideal world, all RNA-seq data would come with the percent mapped reads 
+statistics associated with it. 
+This should be one of the first things you filter your data by.
+Side note: even the highest quality single-cell RNA-seq data still has much 
+lower percent mapped reads than bulk RNA-seq, so don't expect the same range of
+percent mapped reads of bulk RNA-seq.
+However, because a lot of publicly available data does not have percent mapped
+read statistics associated with it, we are going to attempt to get around this
+by trying other strategies of filtering first.
+
+#### Total counts as a quality measure
+
+One basic of method of exploring if a particular cell's data is trustworthy is
+seeing how many counts were measured total, over all the genes. 
+Let's take a look at this data from how many counts are there per sample:
+Note that for `apply` the second argument indicates whether the function you 
+specify is being applied across all the rows or the columns (1 = rows, 2 = columns).
+Also note that in this case we are excluding the `genes` column for the calculation. 
+
+```{r}
+# Make a vector of the total number of counts per sample
+total_counts <- apply(sc_data[, -1], 2, sum)
+
+# Take a look at the summary statistics for the total counts
+summary(total_counts)
+```
+
+Yikes, one of the samples has only 48 counts. 
+It's highly likely that this 'cell' is either an empty well or the well it was
+in didn't get sequenced properly. 
+Regardless of what may have happened, we can't trust that 'cell'.
+These kinds of things are to be expected; it's highly unlikely that when obtain 
+a single-cell RNA-seq dataset that all of cells within the set have reliable, 
+high quality data.
+
+This is why exploring your data and filtering it is important before we attempt to
+look for any biologically meaningful information. Let's visualize the 
+distribution of total counts to see if the 48 count sample is the only one we 
+should get rid of.
+
+For the following graphs, we will use vertical red lines to indicate where 
+maybe a suggested cutoff should go.
+
+```{r Explore total counts distribution}
+# Let's take a look at what this looks like:
+hist(total_counts, col = "lightblue", breaks = 40);
```

Comment:
Any reason for going with base graphics over a `qplot` here?

Reply:
It was a lot quicker and easier to manipulate.

Reply:
I started making it with ggplot2, but this was so much simpler.